python/cudaq/kernel/kernel_decorator.py

```diff
@@ -46,8 +46,18 @@ def __init__(self, function, verbose=False, module=None, kernelName=None):
                         ) if self.kernelFunction is not None else ('', 0)
 
         # Get any global variables from parent scope.
-        self.globalScopedVars = dict(inspect.getmembers(
-            inspect.stack()[2][0]))['f_locals']
+        # We filter only types we accept: integers and floats.
+        # Note here we assume that the parent scope is 2 stack frames up
```

Comment:
I don't think so. This code only gets called when the decorator is constructed, so at the `@cudaq.kernel` call site, but there is one hoop there to jump through, namely the `def kernel(...)` function at https://github.com/NVIDIA/cuda-quantum/blob/experimental%2Fpython/python/cudaq/kernel/kernel_decorator.py#L218. 

So `stack()[0]` is the current stack (inside the constructor), `stack()[1]` has to be the `def kernel(...)` function, and `stack()[2]` has to be the stack containing the parent scope we want access to.

Reply:
Not for this PR - but for my own benefit, do you think it would eventually be desirable to traverse up the stack higher and higher? I'm wondering if users will ever have a desire to pull in something like global variables that aren't considered `f_locals` in a parent stack.

Reply:
I had not thought of that, but yes I think this would be desirable. It may be a very simple adjustment that enables that here. Let me think on it.

Reply:
So turns out this already works 
```python

def test_inner_function_capture():

    n = 3 

    def innerClassical():
        

        @cudaq.kernel()
        def foo():
            q = cudaq.qvector(n)
        
        counts = cudaq.sample(foo) 
        counts.dump()
    
    innerClassical()
```
(It appears that the `f_locals` variables for `innerClassical` also contain those variables from `test_inner_function_capture`, which makes sense) 

Does this address your comment?